controllers/clustertopology_controller_compute.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1alpha4"
+)
+
+// clusterTopologyClass holds all the objects required for computing the desired state of a managed Cluster topology.
+type clusterTopologyClass struct {
+	cluster                                   *clusterv1.Cluster                        //nolint:structcheck
+	clusterClass                              *clusterv1.ClusterClass                   //nolint:structcheck
+	clusterInfrastructureTemplate             *unstructured.Unstructured                //nolint:structcheck
+	clusterControlPlaneTemplate               *unstructured.Unstructured                //nolint:structcheck
+	controlPlaneMachineInfrastructureTemplate *unstructured.Unstructured                //nolint:structcheck
+	machineDeploymentTemplates                map[string]machineDeploymentTopologyClass //nolint:structcheck
```

Comment:
```suggestion
	workersMachineDeploymentTemplates                map[string]machineDeploymentTopologyClass //nolint:structcheck
```
Just a suggestion. Could be considered more consistent with the fields above (and based on the structure of ClusterClass), but I'm not even sure if I like it better.

Reply:
I would prefer to not add workers, it is kind of redundant (we can eventually alway re iterate later when we will actually use those types in the code).